backend/controller/timeline/events_call.go

```diff
@@ -0,0 +1,136 @@
+package timeline
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/types/either"
+	"github.com/alecthomas/types/optional"
+
+	ftlencryption "github.com/TBD54566975/ftl/backend/controller/encryption/api"
+	"github.com/TBD54566975/ftl/backend/controller/timeline/internal/sql"
+	"github.com/TBD54566975/ftl/backend/libdal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type CallEvent struct {
+	ID               int64
+	DeploymentKey    model.DeploymentKey
+	RequestKey       optional.Option[model.RequestKey]
+	ParentRequestKey optional.Option[model.RequestKey]
+	Time             time.Time
+	SourceVerb       optional.Option[schema.Ref]
+	DestVerb         schema.Ref
+	Duration         time.Duration
+	Request          json.RawMessage
+	Response         json.RawMessage
+	Error            optional.Option[string]
+	Stack            optional.Option[string]
+}
+
+func (e *CallEvent) GetID() int64 { return e.ID }
+func (e *CallEvent) event()       {}
+
+// The internal JSON payload of a call event.
+type eventCallJSON struct {
+	DurationMS int64                   `json:"duration_ms"`
+	Request    json.RawMessage         `json:"request"`
+	Response   json.RawMessage         `json:"response"`
+	Error      optional.Option[string] `json:"error,omitempty"`
+	Stack      optional.Option[string] `json:"stack,omitempty"`
+}
+
+type Call struct {
```

Comment:
Why do we need both `Call` and `CallEvent`?

Reply:
The idea here is the `Call` is what's used as the external API to insert calls into the db. And `CallEvent`s are what are returned from the query. Maybe there's a better name for this 🤔

Reply:
Awesome! Will do in the next PR. I might have some questions about how to do this correctly :)